examples/multimodal/utils/audio_loader.py

```diff
@@ -0,0 +1,89 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import logging
+from io import BytesIO
+from typing import Tuple
+from urllib.parse import urlparse
+
+import httpx
+import librosa
+import numpy as np
+
+logger = logging.getLogger(__name__)
+
+
+class AudioLoader:
+    CACHE_SIZE_MAXIMUM = 8
+
+    def __init__(self, cache_size: int = CACHE_SIZE_MAXIMUM):
+        self._http_timeout = 30.0
+        self._http_client = httpx.AsyncClient(timeout=self._http_timeout)
+        self._audio_cache: dict[str, Tuple[np.ndarray, float]] = {}
+        self._cache_queue: asyncio.Queue[str] = asyncio.Queue(maxsize=cache_size)
+
+    async def load_audio(
+        self, audio_url: str, sampling_rate: int = 16000
+    ) -> Tuple[np.ndarray, float]:
+        parsed_url = urlparse(audio_url)
+
+        # For HTTP(S) URLs, check cache first
+        if parsed_url.scheme in ("http", "https"):
+            audio_url_lower = audio_url.lower()
+            if audio_url_lower in self._audio_cache:
+                logger.debug(f"Audio found in cache for URL: {audio_url}")
+                return self._audio_cache[audio_url_lower]
```

Comment:
Could this caching be done by a simpler @lru_cache decorator on the http_client.get?